### PR TITLE
Misc optimizations on typing.

### PR DESCRIPTION
**Pull request by Therzok**

### CODE REVIEW

File: main/src/core/MonoDevelop.Core/MonoDevelop.Core/Properties.cs [hunk 1]
```diff
@@ -103,7 +103,8 @@ TypeConverter GetConverter (Type type)
 		
 		public T Get<T> (string property, T defaultValue)
 		{
-			defaultValues = defaultValues.SetItem (property, defaultValue);
+			if (!defaultValues.Contains (new KeyValuePair<string, object> (property, defaultValue)))
```
Comment (Therzok) [inline main/src/core/MonoDevelop.Core/MonoDevelop.Core/Properties.cs]: I don't know why we do this. Why do we cache the default values (and re-use them in the Get<T> doesn't request a default value?). We should be sane and return a `default(T)` if the overload without a default value is used.

Comment (Therzok) [inline main/src/core/MonoDevelop.Core/MonoDevelop.Core/Properties.cs]: cc @mkrueger @slluis

**Comment by slluis** [comment]
You should separate unrelated optimizations in different PRs. We can't validate the PR if you keep adding changes.

**Comment by Therzok** [comment]
Sorry, last commit wasn't meant to be done on this PR.

File: main/src/core/MonoDevelop.Core/MonoDevelop.Core/Properties.cs (hunk 1 above)
Comment (mkrueger) [inline main/src/core/MonoDevelop.Core/MonoDevelop.Core/Properties.cs]: Has mostly historical reasons - I think the cache is no longer needed and can be removed.

Comment (Therzok) [inline main/src/core/MonoDevelop.Core/MonoDevelop.Core/Properties.cs]: Will change if I see this being time consuming in the future.

**Pull request merged**